src/ai/backend/manager/registry.py

```diff
@@ -295,6 +296,13 @@ async def init(self) -> None:
         evd.consume(
             KernelPullingEvent, self, handle_kernel_creation_lifecycle, name="api.session.kpull"
         )
+        evd.consume(ImagePullStartedEvent, self, handle_image_lifecycle, name="api.session.ipullst")
+        evd.consume(
+            ImagePullFinishedEvent, self, handle_image_lifecycle, name="api.session.ipullfin"
+        )
+        evd.consume(
+            ImagePullFailedEvent, self, handle_image_lifecycle, name="api.session.ipullfail"
+        )
```

Comment:
I have a question.
In the `consume` function, you can already register different handlers for each event, so why is it written to register the same handler function and use switch-case internally?
It seems better to write and apply each handler to handle each event differently.

Reply:
I guess we tried to handle a group of **similar** events in one function. It can reduce code redundancy but as you suggest, we can refactor this to per-event functions.

Reply:
Shouldn't you separate the code here too?

Reply:
Updated this part